datasets/utils.py

```diff
@@ -0,0 +1,21 @@
+import keyword
+
+
+def _is_upper_pascal_case(name: str) -> bool:
+    is_valid: bool = (
```

Comment:
minor: we could use [inflection](https://inflection.readthedocs.io/en/latest/#inflection.camelize) to check for CamelCase

Reply:
I took a look **inflection** first.  Any dependency we take on is taken on by all customers, a goal is to only introduce necessary dependencies.  **inflection** is not widely used, uses regular expressions (which by my experience are very hard to get correct, and there always exists some bug).  
Also checking for camel case is not difficult, and is more maintainable and faster being non regex.

Reply:
I disagree a bit re:regex, manual checks specially for complicated cases can never reach the completeness and ease of mind a regex can give, and can result in spaghetti code, but in this case it's not a big deal

Reply:
I've been burned by difficult to debug regex code in the past...
My bar to move to regex (another language) if really needed...
I agree, in this case it's not a big deal because logic is small and simple.

Reply:
Are we allowing ascii chars in name:  `name.isascii()` ?